### PR TITLE
home work 3

### PR DESCRIPTION
**Pull request by lohikaysapsai**

### CODE REVIEW

File: src/dataStorage.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+export interface IMovie {
+  imdbID?: string
+  name?: string
```
Comment (franxx32) [inline src/dataStorage.ts:3]: I guess name and imdbID  are not optional fields

File: src/index.ts [hunk 2]
```diff
@@ -18,29 +15,68 @@ if (!setConfig(appArgs)) {
...
+
+function errorHandler(error: any, request: any, response: any, next: any) {
+  console.error(`Error: ${error.message}`);
+  response.status(500).send({ error: STATUS_CODES[500] });
+  next();
+}
+
+app.use(express.json());
+
+app.get('/', requestLogger, (req, res) => {
```
Comment (franxx32) [inline src/index.ts:33]: you can set `requestLogger` like global middleware `app.use(requestLogger)`

File: src/index.ts [hunk 3]
```diff
@@ -18,29 +15,68 @@ if (!setConfig(appArgs)) {
...
+function requestLogger(request: any, response: any, next: any) {
+  const { route: { path }, method, body, query, headers } = request;
+  logInfo( `request: ${path} ${method} ${JSON.stringify(body)} ${JSON.stringify(query)} ${JSON.stringify(headers)}`);
+  next();
+}
+
+function errorHandler(error: any, request: any, response: any, next: any) {
+  console.error(`Error: ${error.message}`);
+  response.status(500).send({ error: STATUS_CODES[500] });
+  next();
```
Comment (franxx32) [inline src/index.ts]: Does it make sense to call `next` in errorHandler ?

**Review by franxx32 — APPROVED** [review]

File: src/index.ts (hunk 3 above)
Comment (lohikaysapsai) [reply src/index.ts]: you are right, this is the last middleware function and there is no need to call "next"

**Pull request merged**